lib/rubocop/cop/rails/active_model_errors_direct_manipulation.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks direct manipulation of ActiveModel#errors as hash.
+      # These operations are deprecated in Rails 6.1 and will not work in Rails 7.
+      #
+      # @example
+      #   # bad
+      #   user.errors[:name] << 'msg'
+      #   user.errors.messages[:name] << 'msg'
+      #
+      #   # good
+      #   user.errors.add(:name, 'msg')
+      #
+      #   # bad
+      #   user.errors[:name].clear
+      #   user.errors.messages[:name].clear
+      #
+      #   # good
+      #   user.errors.delete(:name)
+      #
+      class ActiveModelErrorsDirectManipulation < Base
```

Comment:
How about the name `DeprecatedActiveModelErrorsMethods`?

Reply:
If the cop is targeting Rails 6.1 or higher only, can you specify `extend TargetRailsVersion` and `minimum_target_rails_version 6.1` (and add tests)?

- https://github.com/rubocop/rubocop-rails/blob/v2.10.1/lib/rubocop/cop/rails/application_record.rb#L21-L23
- https://github.com/rubocop/rubocop-rails/blob/v2.10.1/spec/rubocop/cop/rails/application_record_spec.rb#L4
- https://github.com/rubocop/rubocop-rails/blob/v2.10.1/spec/rubocop/cop/rails/application_record_spec.rb#L52

Reply:
There are many other methods which are deprecated. This is only the dynamic manipulation part. What do you think of `DeprecatedActiveModelErrorsDynamicManipulation`?

> If the cop is targeting Rails 6.1 or higher only

I recommend running this before upgrading to Rails 6.1, so developers can upgrade the usage, and be ready for Rails 7 in the future. In theory it should be runnable since Rails 3.0.

Reply:
> How about the name DeprecatedActiveModelErrorsMethods?

I originally aligned this with other cops, e.g.: `ActiveRecordCallbacksOrder` and `ActiveSupportAliases`, so I feel maybe we should keep this for consistency?